src/snowflake/cli/api/exceptions.py

```diff
@@ -128,3 +128,13 @@ def __init__(self):
         super().__init__(
             "Schema not specified. Please update connection to add `schema` parameter, or re-run command using `--schema` option. Use `snow connection list` to list existing connections."
         )
+
```

Comment:
Our team has recently tried to avoid introducing too many special exceptions and instead use the generic `ClickException` or its applicable subtype, with an informative message, to raise those exceptions instead. 

That, or maybe reuse/rename [SourceNotFoundError](https://github.com/snowflakedb/snowflake-cli/blob/main/src/snowflake/cli/plugins/nativeapp/artifacts.py#L40) which already throws the error for both file and directory, so you won't need two different exceptions.

Reply:
I agree, renaming `SourceNotFoundError` to `LocalFileNotFoundError` or something could be the cleanest solution. Name up to you.